hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx

```diff
@@ -0,0 +1,337 @@
+
+export const ERROR_MESSAGES = {
+    quantityMissing: "Quantity is required",
+};
+
+const addToCartFormSchema = Yup.object().shape({
+    quantity: Yup.number().required(ERROR_MESSAGES.quantityMissing),
+});
+
+const createQuantityList = (number: number) => {
+    let entry = [];
+
+    for (let i = 1; i <= number; i++) {
+        entry.push(
+            <MenuItem key={i} role="quantity" value={i.toString()}>
+                {i}
+            </MenuItem>
+        );
+    }
+
+    return entry;
+};
+
+interface AddToCartFormProps extends FormikValues {
+    quantityAvailable: number;
+    maxPerTeam?: number;
+}
+export const AddToCartForm = ({
+    quantityAvailable,
+    maxPerTeam,
+    handleSubmit,
+    handleChange,
+    values: { quantity },
+}: AddToCartFormProps) => {
+    const dropdownNum =
+        maxPerTeam !== undefined
+            ? Math.min(quantityAvailable, maxPerTeam)
+            : quantityAvailable;
+
+    return (
+        <>
+            <form className={styles.form} onSubmit={handleSubmit}>
+                <FormControl variant="outlined" className={styles.formControl}>
+                    <InputLabel id="qtyLabel">Qty</InputLabel>
+                    <Select
+                        value={dropdownNum === 0 ? "" : quantity}
+                        onChange={handleChange}
+                        label="Qty"
+                        labelId="qtyLabel"
+                        name="quantity"
+                        disabled={dropdownNum === 0}
+                    >
+                        {createQuantityList(dropdownNum)}
+                    </Select>
+                </FormControl>
+                <div className={styles.formButton}>
+                    <Button
+                        variant="contained"
+                        color="primary"
+                        fullWidth={true}
+                        size="large"
+                        type="submit"
+                        onClick={handleSubmit}
+                        disabled={dropdownNum === 0}
+                        disableElevation
+                    >
+                        Add to cart
+                    </Button>
+                </div>
+            </form>
+        </>
+    );
+};
+
+interface EnhancedAddToCartFormProps {
+    quantityAvailable: number;
+    hardwareId: number;
+    name: string;
+    maxPerTeam?: number;
+}
+export const EnhancedAddToCartForm = ({
+    quantityAvailable,
+    hardwareId,
+    name,
+    maxPerTeam,
+}: EnhancedAddToCartFormProps) => {
+    const dispatch = useDispatch();
+    const currentQuantityInCart =
+        useSelector((state: RootState) => cartSelectors.selectById(state, hardwareId))
+            ?.quantity ?? 0;
+
+    const onSubmit = (formikValues: { quantity: string }) => {
+        const numQuantity: number = parseInt(formikValues.quantity);
+        if (currentQuantityInCart + numQuantity <= (maxPerTeam || quantityAvailable)) {
```

Comment:
Since you default `maxPerTeam` to `Infinity`, I don't think there's any situation where it's undefined, meaning this check would always be true. Also what about maxPerTeam of 0? (No idea why we'd do that, but should handle the constraint still).

You might want to find another way of handling `maxPerTeam` in the outer component other than using `Infinity`, since it gets hard to reason about what's going on.

Reply:
I still use infinity but I initialize maxPerTeam to null

Reply:
```suggestion
        if (currentQuantityInCart + numQuantity <= (maxPerTeam ?? quantityAvailable)) {
```

No more problem with `Infinity`, and we can handle a weird constraint of max 0 per team